embassy-executor-macros/src/macros/main.rs

```diff
@@ -112,17 +125,44 @@ pub fn run(args: TokenStream, item: TokenStream, arch: &Arch) -> TokenStream {
         error(&mut errors, &f.sig, "main function must have 1 argument: the spawner.");
     }
 
-    let entry = match args.entry.as_deref().or(arch.default_entry) {
-        None => TokenStream::new(),
-        Some(x) => match TokenStream::from_str(x) {
+    let entry = match (&args.entry.as_deref(), &arch.default_entry) {
+        (None, Entry::NotRequired) => TokenStream::new(),
+        (Some(x), _) | (None, Entry::Some(x)) if *x == "" => TokenStream::new(),
+        (Some(x), _) | (None, Entry::Some(x)) => match TokenStream::from_str(x) {
             Ok(x) => quote!(#[#x]),
             Err(e) => {
                 error(&mut errors, &f.sig, e);
                 TokenStream::new()
             }
         },
+        (None, Entry::Required) => {
+            error(
+                &mut errors,
+                &f.sig,
+                "An entrypoint macro, or \"\" must be specified when no architecture features are selected",
+            );
+            TokenStream::new()
+        }
     };
 
+    let executor = match (args.executor.as_deref(), &arch.default_entry) {
+        (None, Entry::Required) => {
+            error(
+                &mut errors,
+                &f.sig,
+                "An executor must be specified when no architecture features are selected",
```

Comment:
It's better if the error specifies *how* the user can fix the issue. Something like:

> No architecture selected for `embassy-executor`. Make sure you've enabled one of the `arch-*` features in your `Cargo.toml`.
> 
> Alternatively, if you want to use a custom executor implementation specify it with the `executor` argument. For example: `#[embassy_executor::main(executor = "some_crate::Executor")]`.

Reply:
I wasn't aware multi-line errors from proc macros work as well as they do, nice.